### PR TITLE
Spec systemd

### PR DESCRIPTION
**Pull request by tiran**

### CODE REVIEW

File: custodia.spec [hunk 1]
```diff
@@ -191,6 +192,24 @@ cp %{buildroot}/%{_bindir}/custodia-cli %{buildroot}/%{_bindir}/custodia-cli-3
 %endif
 
 
+%pre
+getent group custodia >/dev/null || groupadd -r custodia
+if ! getent passwd custodia >/dev/null; then
+     useradd -r -g custodia -d / -s /sbin/nologin -c "User for custodia" custodia
+     if [ -d %{_sysconfdir}/custodia ]; then
+         chown -R custodia:custodia %{_sysconfdir}/custodia
+     fi
```
Comment (simo5) [inline custodia.spec]: Why do we need this ?
Shouldn't we just state what the permission should be in install lines ?

If this is for upgrades it should be handled in upgrades only.

File: custodia.spec [hunk 2]
```diff
@@ -169,6 +169,7 @@ mkdir -p %{buildroot}/%{_unitdir}
 mkdir -p %{buildroot}/%{_tmpfilesdir}
 mkdir -p %{buildroot}/%{_localstatedir}/lib/custodia
 mkdir -p %{buildroot}/%{_localstatedir}/log/custodia
+mkdir -p %{buildroot}/%{_localstatedir}/run/custodia
```
Comment (simo5) [inline custodia.spec]: This is not sufficient given the run dir is killed at every reboot, we need to add a tmpfiles.d snippet so that a new dir is created by systemd at startup

Comment (tiran) [inline custodia.spec]: We have tmpfiles.d support. The line is required despite tmpfiles.d support, https://fedoraproject.org/wiki/Packaging:Tmpfiles.d#Example_spec_file

File: custodia.spec (hunk 1 above)
Comment (tiran) [inline custodia.spec]: It's required for existing files that are not directly provided by custodia package, e.g. ``custodia.keytab``. I only want to execute the lines when custodia user does not exist yet. I don't think this can be handled in upgrade.

File: custodia.spec (hunk 2 above)
Comment (tiran) [inline custodia.spec]: ``%{_tmpfilesdir}/custodia.conf`` contains

```
d /run/custodia 0755 custodia custodia
```

Comment (simo5) [inline custodia.spec]: Ops sorry, missed that

File: custodia.spec (hunk 1 above)
Comment (simo5) [inline custodia.spec]: What if a user exist but files are not owned by it ?
mentioning the keytab makes me nervous though, what if the admin had carefully defined permissions (maybe using ACLs or other stuff) ? We would trample over it.

I am thinking whether we should do this automatically at all or rather leave to the admin to set up custodia with a user. Perhaps we could provide an instance creation scripts that offers to create the appropriate user instead of trying to stampeded on existing systems.

Comment (simo5) [inline custodia.spec]: In any case I think we should move user creation etc on it's own PR, this change is important enough to be reviewed separately from other spec fixes.
